glib/src/types.rs

```diff
@@ -100,7 +100,7 @@ impl Type {
     }
 
     #[doc(alias = "g_type_is_a")]
-    pub fn is_a(&self, other: &Type) -> bool {
+    pub fn is_a(&self, other: Self) -> bool {
```

Comment:
Should do the same for all the other functions here, I guess. Take `self` instead of `&self`, etc.

Reply:
Leaving `is_valid` with `&self` made it possible to write `.filter(Type::is_valid)`. On the other hand, `map` would prefer `self`.

Reply:
The reasoning here is mostly https://rust-lang.github.io/rust-clippy/master/index.html#trivially_copy_pass_by_ref

Reply:
Done.